### PR TITLE
Small Fixes in bounded lattice and rational polytope

### PR DESCRIPTION
**Pull request by JacquesOlivierLachaud**
# PR Description

The number of halfspace constraints in BoundedLatticePolytope and BoundedRationalPolytope could be overestimated in some cases. Although it would only create a too big vector for one data member, it is preferable to fix this bug. Furthermore we have added a parameter that allows the user to decide whether or not to check for duplicate halfspace constraints. No unit test added but they will come with the forthcoming Quick Hull pull request.

# Checklist

- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -45,6 +41,13 @@
     set for the Board3DTo2D export (David Coeurjolly,
     [#1537](https://github.com/DGtal-team/DGtal/pull/1537))
 
+- *Geometry*
+  - Small fix and updates in BoundedLatticePolytope and BoundedRationalPolytope
```
Comment (dcoeurjo) [inline ChangeLog.md:45]:
```suggestion
  - Small fixes and updates in BoundedLatticePolytope and BoundedRationalPolytope
```

**Review by dcoeurjo — APPROVED** [review]
Thx @JacquesOlivierLachaud, just a minor comment on the Changelog

**Comment by JacquesOlivierLachaud** [comment]
Ok for the suggestion. What is the new "DGtal-team.DGtal" check ? Is it some kind of review approval ?

**Comment by dcoeurjo** [comment]
It is related to the python binding deployment (from @phcerdan)

**Comment by phcerdan** [comment]
Before the PR pybind11 is merged, you can safely ignore the azure-pipelines checks. Sorry for the noise.

**Comment by JacquesOlivierLachaud** [comment]
Thanks for the info. Looks fine now. Since you already approved the PR, I self merge.

**Pull request merged**